sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/FormRecognizerAsyncClient.java

```diff
@@ -382,18 +384,50 @@ public FormRecognizerServiceVersion getServiceVersion() {
 
     private Function<PollingContext<OperationResult>, Mono<OperationResult>> receiptStreamActivationOperation(
         Flux<ByteBuffer> data, long length, FormContentType formContentType, boolean includeTextDetails) {
-        return (pollingContext) -> {
-            try {
-                Objects.requireNonNull(data, "'data' is required and cannot be null.");
-                Objects.requireNonNull(formContentType, "'formContentType' is required and cannot be null.");
-                return service.analyzeReceiptAsyncWithResponseAsync(ContentType.fromString(formContentType.toString()),
-                    data, length, includeTextDetails)
-                    .map(response -> new OperationResult(
+        Objects.requireNonNull(data, "'data' is required and cannot be null.");
+
+        // Size of buffered data, it used to collect all data from input Flux<ByteBuffer>
+        final long[] bufferedDataSize = {0};
+        // Cached byte buffer
+        final List<ByteBuffer> cachedBuffers = new LinkedList<>();
+        // A boolean value that only used once for guessing content type on input data
+        final AtomicBoolean[] isGuessed = {new AtomicBoolean(true)};
+        // Content type on input data
+        final ContentType[] contentType = new ContentType[1];
+
+        return pollingContext -> data.filter(ByteBuffer::hasRemaining)
+            .windowUntil(buffer -> {
+
+                if (bufferedDataSize[0] >= length) {
+                    return false;
+                }
+                bufferedDataSize[0] += buffer.remaining();
+
+                // if content type is not given, do guessing the content type.
+                if (formContentType == null && isGuessed[0].get() && bufferedDataSize[0] >= 4) {
+                    byte[] bytes = buffer.array();
+                    contentType[0] = getContentType(ByteBuffer.wrap(bytes));
+                    isGuessed[0].compareAndSet(true, false);
+                }
+                // Cache the data to a list of byte buffers
+                ByteBuffer cachedBuffer = ByteBuffer.allocate(buffer.remaining()).put(buffer);
+                cachedBuffer.flip();
+                cachedBuffers.add(cachedBuffer);
+
+                if (bufferedDataSize[0] >= length) {
+                    return true;
+                }
+                return false;
+            }, true, Integer.MAX_VALUE)
```

Comment:
Is this done to support hot fluxes? I thought this API only supported cold fluxes. If we only support cold fluxes, this can be simplified.

Reply:
right but like we discussed offline this could be a cold + self-destructed flux, so cloning is required for scanning of content type, and we don't know upfront that its self-destructed flux or not :( .. that's why i thought we should do windowUntil (and the cloning comes with it) only if contentType is missing.

Reply:
@srnagar - Regarding cold (replay-able) Flux, was thinking a bit more on it. Seeing an issue with self-destructible Flux assumption - though we'll manage to read the header here (for content type) with a single subscription, there is still an issue with retry since each retry is a new subscription unless we do ByteBuffer pooling + back-pressure internally. I don't know this API supports chunked upload where pooling can be applied.

Maybe chunked upload (if supported) could be a future improvement. Keeping that chunk/pooling aside - I think it's better if we make it clear in the doc that this API expects replay-able Flux and go with double subscription, which will anyway happen with retry. Thoughts?

Reply:
My assumption with this API was always that it's going to be a replayable flux. A non-replayable flux would require us to deal with what happens when we want to retry? Storage blob provides a mechanism to have chunk uploads that can be committed once the whole flux is uploaded. Form Recognizer doesn't have this capability. 

So, having a large media file upload requires us to cache the entire flux in-memory or use storage to upload chunks. Also, I am not sure if a non-replayable stream is supported in other languages. If it's not, then we should also make it clear in our documentation that only cold fluxes are supported.

Reply:
yep, I think good to have this assumption added in our Java guidelines, i.e. all APIs that takes `Flux<BuyteBuffer>` requires the stream to be replay-able. 

If form recognizer support recognizing content of a blob then its good to have a note in the FR api that takes `Flux<BuyteBuffer>`that - for large files use storage optimized-reliable upload and use relevant FR api that takes blob uri.